sensor/common/detector/enricher.go

```diff
@@ -170,6 +179,55 @@ func (c *cacheValue) scanAndSet(ctx context.Context, svc v1.ImageServiceClient,
 	c.image = scannedImage.GetImage()
 }
 
+func (c *cacheValue) scanAndSetWithLock(ctx context.Context, svc v1.ImageServiceClient, req *scanImageRequest) {
+	defer c.signal.Signal()
+
+	// A cacheValue is unique per image, obtain the lock before scanning to avoid sending multiple
+	// scan requests for the same image name at the same time.
+	c.lock.Lock()
+	defer c.lock.Unlock()
```

Comment:
Do we add this `cacheValue` to the cache before scanning? I am thinking of the scenario of each scan request having its value until one of them is added to the cache, and then this lock would effectively function.

Reply:
Yes it is added to the cache before scanning with this line: 

https://github.com/stackrox/stackrox/blob/84014bb2026db2a7edeb6945f29eeaa9b25dee20/sensor/common/detector/enricher.go#L234